pkg/scheduler/framework/v1alpha1/types.go

```diff
@@ -110,12 +118,44 @@ func getAffinityTerms(pod *v1.Pod, v1Terms []v1.PodAffinityTerm) []AffinityTerm
 	return terms
 }
 
+// getWeightedAffinityTerms returns the list of processed affinity terms.
+func getWeightedAffinityTerms(pod *v1.Pod, v1Terms []v1.WeightedPodAffinityTerm) []WeightedAffinityTerm {
+	if v1Terms == nil {
+		return nil
+	}
+
+	var terms []WeightedAffinityTerm
+	for _, term := range v1Terms {
+		t := newAffinityTerm(pod, &term.PodAffinityTerm)
+		if t == nil {
+			// We get here if the label selector failed to process, this is not supposed
+			// to happen because the pod should have been validated by the api server.
+			return nil
+		}
+		terms = append(terms, WeightedAffinityTerm{AffinityTerm: *t, Weight: term.Weight})
+	}
+	return terms
+}
+
 // NewPodInfo return a new PodInfo
 func NewPodInfo(pod *v1.Pod) *PodInfo {
+	var preferredAffinityTerms []v1.WeightedPodAffinityTerm
+	var preferredAntiAffinityTerms []v1.WeightedPodAffinityTerm
+	if affinity := pod.Spec.Affinity; affinity != nil {
+		if a := affinity.PodAffinity; a != nil {
+			preferredAffinityTerms = a.PreferredDuringSchedulingIgnoredDuringExecution
+		}
+		if a := affinity.PodAntiAffinity; a != nil {
+			preferredAntiAffinityTerms = a.PreferredDuringSchedulingIgnoredDuringExecution
+		}
+	}
+
 	return &PodInfo{
-		Pod:                       pod,
-		RequiredAffinityTerms:     getAffinityTerms(pod, schedutil.GetPodAffinityTerms(pod.Spec.Affinity)),
-		RequiredAntiAffinityTerms: getAffinityTerms(pod, schedutil.GetPodAntiAffinityTerms(pod.Spec.Affinity)),
+		Pod:                        pod,
+		RequiredAffinityTerms:      getAffinityTerms(pod, schedutil.GetPodAffinityTerms(pod.Spec.Affinity)),
+		RequiredAntiAffinityTerms:  getAffinityTerms(pod, schedutil.GetPodAntiAffinityTerms(pod.Spec.Affinity)),
+		PreferredAffinityTerms:     getWeightedAffinityTerms(pod, preferredAffinityTerms),
```

Comment:
nit: add an arg (to tell required or preferred) to `schedutil.GetPodAntiAffinityTerms` so we don't need L142~L151.

Reply:
we will need this logic somewhere, either here or inside the other function. Having it here will be shorter because there we will have to check the arg.

Reply:
Uhm... this inconsistency between preferred and required bothers me a bit. But we can get rid of it once we remove the dependency from queue to `GetPodAffinityTerms`.

Other than that, it might be simpler to do:

```
podInfo := &PodInfo{Pod: pod}
```

and then fill up the rest from within the conditionals checking for `!= nil`

Reply:
```GetPodAffinityTerms``` and  ```GetPodAntiAffinityTerms``` doesn't just check for nil, but also encapsulates the now commented out logic to append the ```*RequiredDuringExecution``` terms. Let's merge this as is and move all those checks here as you suggest once the queue's dependency on ```GetPodAffinityTerms``` is removed.